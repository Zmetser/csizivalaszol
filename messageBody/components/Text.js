/**
 * @flow
 * 
 * Create formatted text elements from TextNode structures.
 */

import { createElement } from 'react'
import type { TextNode, InlineStyle } from '../types'

function inlineStyleToTag (styleName: InlineStyle): string {
  switch (styleName) {
    case 'Italic': return 'em'
    case 'Bold': return 'strong'
    case 'Struck': return 's'
    case 'Small': return 'small'
    case 'Code': return 'code'
    case 'Subscript': return 'sub'
    case 'Superscript': return 'sup'
    case 'Underline': return 'u'
    default: return 'span'
  }
}

// NOTE: For whatever reason I get the Error: `'TextNode' is defined but never used  no-unused-vars`
//       If TextNode is assigned as the props type as is, this proxying however satisfies eslint.
type Props = TextNode
export default function Text ({ value, styles }: Props, key: string | number): React$Element<*> | string {
  const reversedStyles = styles.slice().reverse()
  return reversedStyles.reduce((prev, styleName, index) =>
    createElement(inlineStyleToTag(styleName), { key: styleName + key }, prev), value)
}
