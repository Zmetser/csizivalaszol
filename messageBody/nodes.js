/**
 * @flow
 *
 * All the usable nodes
 */

export type InlineStyle = 'Italic' | 'Bold' | 'Struck' | 'Small' | 'Code' | 'Subscript' | 'Superscript' | 'Underline' | 'Normal'

const TextNode = (value: string, styles: Array<InlineStyle>) => ({
  type: 'Text',
  value,
  styles
})

const LineBreak = () => ({ type: 'LineBreak' })

const Link = (value: Array<TextNode>, href: string, target?: ?string) => ({
  type: 'Link',
  value,
  href,
  target
})

export type InlineNode = TextNode | LineBreak | Link;

type ImageAttributes = { src: string, alt: string, width: number, height: number }
const Image = ({ src, alt, width, height }: ImageAttributes) => ({
  type: 'Image',
  src,
  alt,
  width,
  height
})

const Paragraph = (value: Array<InlineNode>) => ({
  type: 'Paragraph',
  value
})

module.exports = {
  TextNode,
  Paragraph,
  LineBreak,
  Link,
  Image
}
