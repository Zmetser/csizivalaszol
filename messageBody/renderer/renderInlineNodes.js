/**
 * @flow
 */

import * as React from 'react'
import { UnknownNodeError } from '../errors'

import type { InlineNode } from '../types'

import {
  Text,
  Link
} from '../components'

function createInlineNode (node: InlineNode, index): React.Element<*> {
  switch (node.type) {
    case 'Text': return Text(node, index)
    case 'Link': return <Link {...node} key={index} />
    default: throw new UnknownNodeError(node)
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>): Array<React.Element<*>> {
  return nodes.map(createInlineNode)
}
