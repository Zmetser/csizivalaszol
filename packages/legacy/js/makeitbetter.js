function del() {
  $elem = document.getElementById('page');
  $elem.value = "";
}