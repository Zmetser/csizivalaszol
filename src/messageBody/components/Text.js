/**
 * @flow
 *
 * Create formatted text elements from TextNode structures.
 */

import * as React from 'react'
import type { TextNode, InlineStyle } from '../types'

function inlineStyleToTag (styleName: InlineStyle): string {
  switch (styleName) {
    case 'Italic': return 'em'
    case 'Bold': return 'strong'
    case 'Struck': return 's'
    case 'Small': return 'small'
    case 'Code': return 'code'
    case 'Subscript': return 'sub'
    case 'Superscript': return 'sup'
    case 'Underline': return 'u'
    default: return 'span'
  }
}

type Props = { node: TextNode }
export default function Text ({ node }: Props): React.Element<*> | string {
  const { value, styles = [] } = node
  const reversedStyles = styles.slice().reverse()

  return reversedStyles.reduce((prev, styleName, index) =>
    React.createElement(inlineStyleToTag(styleName), { key: styleName }, prev), value)
}
