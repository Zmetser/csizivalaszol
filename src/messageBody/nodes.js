/**
 * All the usable nodes
 */

import type {
  ImageAttributes,
  ImageNode,
  InlineNode,
  InlineStyle,
  LineBreakNode,
  LinkNode,
  ParagraphNode,
  TextNode,
  ListType,
  ListItem,
  ListNode
} from './types'

export const createTextNode = (value: string, styles: Array<InlineStyle>): TextNode => ({
  type: 'Text',
  value,
  styles
})

export const createLineBreak = (): LineBreakNode => ({ type: 'LineBreak' })

export const createLink = (value: Array<TextNode>, href: string, target?: ?string): LinkNode => ({
  type: 'Link',
  value,
  href,
  target
})

export const createImage = ({ src, alt, width, height }: ImageAttributes): ImageNode => ({
  type: 'Image',
  src,
  alt,
  width,
  height
})

export const createParagraph = (value: Array<InlineNode>): ParagraphNode => ({
  type: 'Paragraph',
  value
})

export function createList<T: ListType> (items: Array<ListItem>, listType: T): ListNode<T> {
  return {
    type: 'List',
    listType,
    items
  }
}
