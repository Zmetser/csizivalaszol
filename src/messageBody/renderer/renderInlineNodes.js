/**
 * @flow
 */

import * as React from 'react'
import { UnknownNodeError } from '../errors'

import type { InlineNode } from '../types'

import {
  Text,
  Link,
  LineBreak
} from '../components'

function createInlineNode (node: InlineNode, index): React.Element<*> {
  switch (node.type) {
    case 'Text': return <Text node={node} key={index} />
    case 'Link': return <Link node={node} key={index} />
    case 'LineBreak': return <LineBreak key={index} />
    default: throw new UnknownNodeError(node)
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>): Array<React.Element<*>> {
  return nodes.map(createInlineNode)
}
