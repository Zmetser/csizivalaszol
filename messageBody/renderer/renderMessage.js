/**
 * @flow
 */

import * as React from 'react'
import { Paragraph } from '../components'

import type { BlockNode } from '../types.js'
import { UnknownNodeError } from '../errors'

export default function render (message: Array<BlockNode>): Array<React.Element<*>> {
  return message.map((node, index): React.Element<*> => {
    switch (node.type) {
      case 'Paragraph': return Paragraph({ value: node.value }, index)
      default: throw new UnknownNodeError(node)
    }
  })
}
