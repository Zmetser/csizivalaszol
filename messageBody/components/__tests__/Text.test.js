import renderer from 'react-test-renderer'

import { createTextNode } from '../../nodes.js'

import Text from '../Text'

describe('<Text />', () => {
  test('render without formatting', () => {
    const element = Text(createTextNode('Foo', []))
    expect(element).toBe('Foo')
  })

  test('apply styles', () => {
    const element = renderer.create(
      Text(createTextNode('Foo', ['Bold']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })

  test('apply nested styles', () => {
    const element = renderer.create(
      Text(createTextNode('Foo', ['Bold', 'Italic']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })

  test('unknown style falls back to span', () => {
    const element = renderer.create(
      Text(createTextNode('Foo', ['Bold', 'Foo']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })

  test('text with "Italic" style renders with the correct element', () => {
    const element = renderer.create(
      Text(createTextNode('Italic', ['Italic']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
  test('text with "Bold" style renders with the correct element', () => {
    const element = renderer.create(
      Text(createTextNode('Bold', ['Bold']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
  test('text with "Struck" style renders with the correct element', () => {
    const element = renderer.create(
      Text(createTextNode('Struck', ['Struck']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
  test('text with "Small" style renders with the correct element', () => {
    const element = renderer.create(
      Text(createTextNode('Small', ['Small']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
  test('text with "Code" style renders with the correct element', () => {
    const element = renderer.create(
      Text(createTextNode('Code', ['Code']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
  test('text with "Subscript" style renders with the correct element', () => {
    const element = renderer.create(
      Text(createTextNode('Subscript', ['Subscript']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
  test('text with "Superscript" style renders with the correct element', () => {
    const element = renderer.create(
      Text(createTextNode('Superscript', ['Superscript']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
  test('text with "Underline" style renders with the correct element', () => {
    const element = renderer.create(
      Text(createTextNode('Underline', ['Underline']))
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
})
