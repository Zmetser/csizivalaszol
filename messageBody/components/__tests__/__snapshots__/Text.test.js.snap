// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> apply nested styles 1`] = `
<strong>
  <em>
    Foo
  </em>
</strong>
`;

exports[`<Text /> apply styles 1`] = `
<strong>
  Foo
</strong>
`;

exports[`<Text /> text with "Bold" style renders with the correct element 1`] = `
<strong>
  Bold
</strong>
`;

exports[`<Text /> text with "Code" style renders with the correct element 1`] = `
<code>
  Code
</code>
`;

exports[`<Text /> text with "Italic" style renders with the correct element 1`] = `
<em>
  Italic
</em>
`;

exports[`<Text /> text with "Small" style renders with the correct element 1`] = `
<small>
  Small
</small>
`;

exports[`<Text /> text with "Struck" style renders with the correct element 1`] = `
<s>
  Struck
</s>
`;

exports[`<Text /> text with "Subscript" style renders with the correct element 1`] = `
<sub>
  Subscript
</sub>
`;

exports[`<Text /> text with "Superscript" style renders with the correct element 1`] = `
<sup>
  Superscript
</sup>
`;

exports[`<Text /> text with "Underline" style renders with the correct element 1`] = `
<u>
  Underline
</u>
`;

exports[`<Text /> unknown style falls back to span 1`] = `
<strong>
  <span>
    Foo
  </span>
</strong>
`;
